### PR TITLE
Add a new "failed" task status for skipped vaccines dosages

### PR DESCRIPTION
**Pull request by qaziabubakar-vd**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

There is a scenario in one of the application where we need to enable the subsequent dosages of a vaccine to be enabled and in "Ready" status even if the previous dosage of that vaccine was not filled out. To show the dosage with a "clear (x)" icon in the profile, a new "failed" status is integrate to show a dosage was not given. 

**Engineer Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [x] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by ellykits** [comment]
Looks like a great addition.

File: android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirResourceUtil.kt [hunk 1]
```diff
@@ -206,7 +206,10 @@ constructor(
     this.partOf
       .find { it.reference.startsWith(ResourceType.Task.name + "/") }
       ?.let {
-        defaultRepository.fhirEngine.get<Task>(it.extractId()).status.isIn(TaskStatus.COMPLETED)
+        defaultRepository.fhirEngine
+          .get<Task>(it.extractId())
+          .status
+          .isIn(TaskStatus.COMPLETED, TaskStatus.FAILED)
```
Comment (ellykits) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirResourceUtil.kt:212]: Can you write a test for this?

**Review by ellykits — CHANGES_REQUESTED** [review]

File: android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirResourceUtil.kt (hunk 1 above)
Comment (qaziabubakar-vd) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirResourceUtil.kt:212]: Sure, I'll add a test

**Review by f-odhiambo — APPROVED** [review]
LGTM

**Pull request merged**